### PR TITLE
Don't award rep to exercism-bot and exercism-ghost

### PR DESCRIPTION
**Pull request by ErikSchierboom**
- Don't create reputation tokens for system or ghost user
- Don't award PR rep to exercism-ghost and exercism-bot users

Closes #109

### CODE REVIEW

**Review by iHiD — APPROVED** [review]

File: app/commands/user/reputation_token/award_for_pull_request_author.rb [hunk 1]
```diff
@@ -28,7 +28,7 @@ def call
           external_url: params[:html_url],
           merged_at: params[:merged_at]
         )
-        token.update!(level: reputation_level)
+        token&.update!(level: reputation_level)
```
Comment (ErikSchierboom) [inline app/commands/user/reputation_token/award_for_pull_request_author.rb:31]: I had to modify this as the `User::ReputationToken::Create` call above returns `nil` for the system/ghost user.

File: app/commands/user/reputation_token/award_for_pull_request_merger.rb [hunk 2]
```diff
@@ -28,7 +28,7 @@ def call
           external_url: params[:html_url],
           merged_at: params[:merged_at]
         )
-        token.update!(level: reputation_level)
+        token&.update!(level: reputation_level)
```
Comment (ErikSchierboom) [inline app/commands/user/reputation_token/award_for_pull_request_merger.rb:31]: See above

File: app/commands/user/reputation_token/award_for_pull_request_reviewers.rb [hunk 3]
```diff
@@ -35,7 +35,7 @@ def call
             merged_at: params[:merged_at],
             closed_at: params[:closed_at]
           )
-          token.update!(level: reputation_level)
+          token&.update!(level: reputation_level)
```
Comment (ErikSchierboom) [inline app/commands/user/reputation_token/award_for_pull_request_reviewers.rb:38]: See above

**Review by ErikSchierboom — COMMENTED** [review]
Is is true that the only place where a new `User::ReputationPeriod` row is created, is in `User::ReputationPeriod::MarkForNewToken`? (see https://github.com/exercism/website/blob/main/app/commands/user/reputation_period/mark_for_new_token.rb#L21) If so, I probably don't have to do anything special to not create those reputation period tokens for the system/bot user, as the only place the `MarkForNewToken` command is called, is in `User::ReputationToken::Create`, which early returns for the system/ghost users: https://github.com/exercism/website/blob/66144684f307838973066a129b56c315c64daf07/app/commands/user/reputation_token/create.rb#L19

**Comment by iHiD** [comment]
>  If so, I probably don't have to do anything special to not create those reputation period tokens for the system/bot user, as the only place the MarkForNewToken command is called, is in User::ReputationToken::Create, which early returns for the system/ghost users:

Agreed.

**Review by iHiD — APPROVED** [review]

**Comment by iHiD** [comment]
Really nice PR.

**Comment by ErikSchierboom** [comment]
Thanks

**Pull request merged**